docs/source/api/apollo-server.md

```diff
@@ -1,195 +1,491 @@
+For an example, see [Define your GraphQL schema](../getting-started/#step-3-define-your-graphql-schema).
+</td>
+</tr>
 
-    Additionally, this `logger` will be made available on the `GraphQLRequestContext` and available to plugins.  This allows a plugin to, e.g., augment the logger on a per-request basis within the `requestDidStart` life-cycle hook.
 
-  * `rootValue`: <`Any`> | <`Function`>
+<tr class="required">
+<td>
 
-    A value or function called with the parsed `Document`, creating the root value passed to the graphql executor. The function is useful if you wish to provide a different root value based on the query operation type.
+###### `resolvers`
 
-```js
-new ApolloServer({
-  typeDefs,
-  resolvers,
-  rootValue: (documentAST) => {
-    const op = getOperationAST(documentNode)
-    return op === 'mutation' ? mutationRoot : queryRoot;
-  }
-});
-```
+`Object` or `Array`
+</td>
+<td>
 
-* `mocks`: <`Object`> | <`Boolean`>
+**Required.** A map of functions that populate data for individual schema fields. Can also be an array of multiple maps that are merged.
 
-  A boolean enabling the default mocks or object that contains definitions
+For details, see [Resolvers](../data/resolvers/).
+</td>
+</tr>
 
-* `mockEntireSchema`: <`Boolean`>
 
-  A boolean controlling whether existing resolvers are overridden by mocks. Defaults to true, meaning that all resolvers receive mocked values.
+<tr>
+<td>
 
-* `schemaDirectives`: <`Object`>
+###### `context`
 
-  Contains definition of schema directives used in the `typeDefs`
+`Object` or `Function`
+</td>
+<td>
 
-* `introspection`: <`Boolean`>
+An object (or a function that _creates_ an object) that's passed to every resolver that executes for a particular operation. This enables resolvers to share helpful context, such as a database connection.
 
-  Enables and disables schema introspection. Disabled in production by default.
+[Certain fields](#middleware-specific-context-fields) are added to this object automatically, depending on which [Node.js middleware](../integrations/middleware/) your server uses.
 
-* `playground`: <`Boolean`> | <`Object`>
+For more details, see [The `context` argument](../data/resolvers/#the-context-argument).
+</td>
+</tr>
 
-  Enables and disables playground and allows configuration of GraphQL Playground. The options can be found on GraphQL Playground's [documentation](https://github.com/prismagraphql/graphql-playground/#usage)
 
-* `debug`: <`Boolean`>
+<tr>
+<td>
 
-  Enables and disables development mode helpers. Defaults to `true`
+###### `introspection`
 
-* `validationRules`: <`Object`>
+`Boolean`
+</td>
 
-  Schema validation rules
+<td>
 
-* `tracing`, `cacheControl`: <`Boolean`>
```

Comment:
I think you lost these?

Personally I don't mind if we pretend `tracing` doesn't exist. I'm pretty sure people do need to set the cacheControl options like `defaultMaxAge` pretty frequently though.